parser/parser/cc/lexer.rl

```diff
@@ -3047,6 +3054,7 @@ void lexer::rewind_and_reset_to_expr_end(size_t newPos) {
 
   // rewind
   this->_p = this->source_buffer.data() + newPos;
+  token_queue.clear();
```

Comment:
These two `.clear()` calls happen because if we reached the end of the file and have decided to reset `p` back to some earlier point in the file (backtrack), any tokens that happened to be in the queue don't make sense anymore--we should entirely start lexing tokens from the new `p`.